source/extensions/filters/http/common/factory_base.h

```diff
@@ -55,6 +58,11 @@ class FactoryBase : public Server::Configuration::NamedHttpFilterConfigFactory {
   createRouteSpecificFilterConfigTyped(const RouteConfigProto&,
                                        Server::Configuration::ServerFactoryContext&,
                                        ProtobufMessage::ValidationVisitor&) {
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.check_unsupported_typed_per_filter_config")) {
+      throw EnvoyException(fmt::format(
+          "The filter {} doesn't support virtual host-specific configurations", name()));
+    }
```

Comment:
Logging a warning here might be useful here to draw attention to a configuration that needs to be fixed.

Reply:
got it, thanks!